homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -98,7 +101,10 @@ def device_state_attributes(self):
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
-
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        if state:
+            self._state = state.state
```

Comment:
We should only restore the last state if we're in manual mode. Otherwise it has already done a speedtest and we should use that data ?